crates/router/src/connector/aci/transformers.rs

```diff
@@ -141,7 +157,25 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for AciPaymentsRequest {
                 card_cvv: ccard.card_cvc,
             })),
             api::PaymentMethodData::PayLater(_) => PaymentDetails::Klarna,
-            api::PaymentMethodData::Wallet(_) => PaymentDetails::Wallet,
+            api::PaymentMethodData::Wallet(ref wallet_data) => match wallet_data {
+                api_models::payments::WalletData::MbWay(data) => {
+                    PaymentDetails::Wallet(Box::new(WalletPMData {
+                        payment_brand: PaymentBrand::Mbway,
+                        account_id: Some(data.telephone_number.clone()),
+                        shopper_result_url: item.request.router_return_url.clone(),
+                    }))
+                }
+                api_models::payments::WalletData::AliPay { .. } => {
+                    PaymentDetails::Wallet(Box::new(WalletPMData {
+                        payment_brand: PaymentBrand::Alipay,
+                        account_id: None,
+                        shopper_result_url: item.request.router_return_url.clone(),
+                    }))
```

Comment:
Is `account_id` being `None` in case of `AliPay` intentional?

Reply:
Yes, account_id is not required for Alipay but a mandatory field for Mbway

Reply:
Cool, marking this conversation resolved.